application/libraries/REST_Controller.php

```diff
@@ -864,40 +864,38 @@ protected function _log_request($authorized = FALSE)
     }
 
     /**
-     * Limiting requests
-     * Check if the requests are coming in a tad too fast.
+     * Check if the requests to a controller method exceed a limit
      *
      * @access protected
      *
-     * @param  string $controller_method The method being called.
+     * @param  string $controller_method The method being called
      *
-     * @return bool
+     * @return bool TRUE the call limit is below the threshold; otherwise, FALSE
      */
     protected function _check_limit($controller_method)
     {
         // They are special, or it might not even have a limit
         if (!empty($this->rest->ignore_limits) || !isset($this->methods[$controller_method]['limit']))
         {
-            // On your way sonny-jim.
+            // Everything is fine
             return TRUE;
         }
 
-        // How many times can you get to this method an hour?
+        // How many times can you get to this method in an hour?
         $limit = $this->methods[$controller_method]['limit'];
 
-        // Get data on a keys usage
+        // Get data about a keys' usage and limit to one row
         $result = $this->rest->db
             ->where('uri', $this->uri->uri_string())
             ->where('api_key', $this->rest->key)
-            ->get(config_item('rest_limits_table'))
-            ->row();
+            ->limit(1)
```

Comment:
I would suggest not doing a limit, and instead look to do a sum that way it would catch multiple entries (if for some reason that happened).